### PR TITLE
MULE-14849: DuplicateExportedPackageException: There are multiple art…

### PR DESCRIPTION
**Pull request by gsfernandes**
…ifacts exporting the same package

### CODE REVIEW

File: modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/ApplicationStartedSplashScreen.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
  */
 public class ApplicationStartedSplashScreen extends ArtifactStartedSplashScreen<ApplicationDescriptor> {
 
+  private static final String PLUGIN_VALUE_FORMAT = " - %s (%s)";
+
   @Override
   public void createMessage(ApplicationDescriptor descriptor) {
     doBody(format("Started app '%s'", descriptor.getName()));
```
Comment (gsfernandes) [inline modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/ApplicationStartedSplashScreen.java]: Should we also log the domain reference?

File: modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultApplicationFactory.java [hunk 2]
```diff
@@ -126,6 +127,11 @@ public Application createArtifact(ApplicationDescriptor descriptor) throws IOExc
         pluginDependenciesResolver.resolve(domain.getDescriptor().getPlugins(),
                                            new ArrayList<>(getArtifactPluginDescriptors(descriptor)));
 
+    // Refreshes the list of plugins on the descriptor with the resolved from domain and transitive plugin dependencies
+    Set resolvedArtifactPlugins = new LinkedHashSet<>();
+    resolvedArtifactPlugins.addAll(resolvedArtifactPluginDescriptors);
+    descriptor.setPlugins(resolvedArtifactPlugins);
```
Comment (gsfernandes) [inline modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultApplicationFactory.java]: I don't like this as it is modifying the descriptor, other approach would be to have a resolvedPlugins field and set the list of plugins descriptor as they were resolved (using semver and also depending on what are the plugins already defined at the domain level)

Comment (elrodro83) [reply modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultApplicationFactory.java]: Should at least log a warning if the transitive dependencies versions are inconsistent, IMO

File: modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/internal/plugin/BundlePluginDependenciesResolver.java [hunk 3]
```diff
@@ -233,6 +252,23 @@ private ClassLoaderModelBuilder createBuilderWithoutExportedPackages(ClassLoader
     return classLoaderModelBuilder;
   }
 
+  private ClassLoaderModelBuilder createBuilderWithoutDependency(ClassLoaderModel originalClassLoaderModel,
+                                                                 BundleDependency dependencyToBeExcluded) {
+    ClassLoaderModelBuilder classLoaderModelBuilder = new ClassLoaderModelBuilder()
+        .dependingOn(originalClassLoaderModel.getDependencies().stream()
+            .filter(dependency -> !dependency.equals(dependencyToBeExcluded)).collect(
+                                                                                      Collectors
+                                                                                          .toCollection(() -> new LinkedHashSet<>())))
```
Comment (elrodro83) [inline modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/internal/plugin/BundlePluginDependenciesResolver.java]: why not just `toSet()` collector?

Comment (gsfernandes) [reply modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/internal/plugin/BundlePluginDependenciesResolver.java]: Order matters here, as later these dependencies will end up creating the classloaders for each plugin and later the extension models are generated for each plugin (classloader) and it is expected that the plugins list is sorted by dependencies as extension models require context to resolve things (for instance: http cannot be loaded if extension model for sockets was not loaded before).

File: modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultApplicationFactory.java (hunk 2 above)
Comment (gsfernandes) [reply modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultApplicationFactory.java]: Something like:

**Pull request merged**